packages/web/src/components/admin_page/feature_switch_page/render_switch.jsx

```diff
@@ -0,0 +1,178 @@
+            [name]: name === 'is_enabled' ? value === 'true' : value,
+        }));
+    };
+
+    const handleSubmit = e => {
+        e.preventDefault();
+        const trimmedSwitchId = newSwitch.switch_id.trim();
+
+        if (trimmedSwitchId === '') {
+            toast.error('Please enter switch name');
+        } else if (trimmedSwitchId.length > 10) {
+            toast.error('Switch name must be 10 characters or less');
+        } else {
+            newSwitch.switch_id = trimmedSwitchId;
+            onUpdate(newSwitch);
+        }
+    };
+
+    return (
+        <Form onSubmit={handleSubmit}>
+            <Form.Group className="mb-3" controlId="formBasicEmail">
+                <Form.Label>Set a feature:</Form.Label>
+                <Form.Control
+                    type="text"
+                    placeholder="Enter new feature switch"
+                    onChange={handleChange}
+                    name="switch_id"
+                    value={newSwitch.switch_id}
+                />
+                <Form.Text className="text-muted">Add a new feature</Form.Text>
+            </Form.Group>
+            <Form.Group className="mb-3" controlId="formBasicIsEnabled">
+                <Form.Label>Set it to true or false:</Form.Label>
+                <Form.Select onChange={handleChange} name="is_enabled" value={newSwitch.is_enabled.toString()}>
+                    <option value="true">True</option>
+                    <option value="false">False</option>
+                </Form.Select>
+                <Form.Text className="text-muted"></Form.Text>
+            </Form.Group>
+
+            <Button type="submit" variant="primary">
+                Add
+            </Button>
+        </Form>
+    );
+};
+
+export const popover = (onUpdate, errorMessage) => {
+    console.log(errorMessage);
+    return (
+        <Popover id="popover-basic">
+            <Popover.Header as="h3">Want to update or remove?</Popover.Header>
+            <Popover.Body>
+                {errorMessage ? <Alert variant="warning">{errorMessage}</Alert> : null}
+                <UpdateSwitch onUpdate={onUpdate} />
+            </Popover.Body>
+        </Popover>
+    );
+};
+
+// Creates a new object updateSwitches that makes a copy of switches and updates a specific switch's is_enabled property
+const updateSwitches = (switches, setSwitches) => data => {
+    setSwitches({ ...switches, [data.switch_id]: data.is_enabled });
+};
+
+//creates a new object updatedSwitches that is equal to switches with the siwtchToRemove, removed
+const deleteSwitches = (switches, setSwitches) => switchToRemove => {
+    // eslint-disable-next-line no-unused-vars
+    const { [switchToRemove]: _, ...updatedSwitches } = switches;
+    setSwitches(updatedSwitches);
+};
+
+const RenderSwitches = () => {
+    //We need to pass what is selected to the child element to track if a switch is selected so that we dont display two messages for editing and adding switch
```

Comment:
<img width="723" alt="image" src="https://github.com/AUBGTheHUB/monolith/assets/104720011/191c1161-b654-448f-bce1-8d0a71f2aeb3">

We need to pass the selected switch (switch_id) as a prop to children components in order to maintain state information about whether a switch is currently selected. This allows us to avoid displaying two popovers (menu's for editing or creating switches) at the same time.